### PR TITLE
[DOCS] Update README.md

### PR DESCRIPTION
**Pull request by leemthompo**
Spruce up the readme with some more helpful context and links.

[📸 Preview](https://github.com/elastic/connectors-python/blob/d51c2395274056612f8ae9081f4e8df733f892ea/README.md)

## `TODO`

- Check backport labels

### CODE REVIEW

**Review by artem-shelkovnikov — DISMISSED** [review]
Looks great! Do we want to also reference running connectors in Docker readme file here?

**Comment by leemthompo** [comment]
> Do we want to also reference running connectors in Docker readme file here?

Yes! I expect we'll iterate on this doc and the overall connector docs a lot in the near future, but I agree we should have quicklinks to the top docs here.

File: README.md [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+These connectors are available as:
+- [**Connector clients**](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html) to be self-managed on your own infrastructure
+- [**Native connectors**](https://www.elastic.co/guide/en/enterprise-search/current/native-connectors.html) using our fully managed service on Elastic Cloud
+
+ℹ️ For an overview of the steps involved in deploying connector clients refer to [**Connector clients**](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
+You can get started quickly with Docker, using [these instructions](docs/DOCKER.md).
+We also have an end-to-end tutorial using the [PostgreSQL connector client](https://www.elastic.co/guide/en/enterprise-search/current/postgresql-connector-client-tutorial.html).
+
+
+## Python connector framework
```
[the remark this answers is not in the document]
Comment (leemthompo) [reply README.md:19]: @seanstory is that not the "connector service"?

[the remark this answers is not in the document]
Comment (seanstory) [reply README.md:19]: In my mind (all of this is my personal opinion/mindset) the "connector service" is what we call it when it's running. So we could call it the "connector service code" or the "connector service libraries" when it's in a repo?

[the remark this answers is not in the document]
Comment (leemthompo) [reply README.md:19]: "connector service code" sounds good to me. I've been using "connector code" from time to time...

[the remark this answers is not in the document]
Comment (leemthompo) [reply README.md:35]: @seanstory I've further cleaned up this page in the latest two commits, plus I've tried to mention the concept of the shared connector service code. Let me know what you think about the readme at this stage!

File: README.md [hunk 2]
```diff
@@ -1,14 +1,52 @@
...
+- [Connector clients](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html)
+- [Native connectors](https://www.elastic.co/guide/en/enterprise-search/current/native-connectors.html)
+
+You'll also find the individual references for each connector there.
+For everything to do with _developing_ connectors, you'll find that here in this repo.
+
+## Connector service code
+
+In addition to the source code for individual connectors, this repo also contains the connector service code, which is used for tasks like running connectors, and managing scheduling, syncs, and cleanup.
+This is shared code that is used by all connectors.
```
Comment (seanstory) [inline README.md:33]:
```suggestion
This is shared code that is not used by individual connectors, but helps to coordinate and run a deployed instance/process.
```

**Review by seanstory — APPROVED** [review]

**Comment by leemthompo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.10|[](https://github.com/elastic/connectors-python/pull/1550)|
|✅|8.9|[](https://github.com/elastic/connectors-python/pull/1551)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**